torch/lib/THC/THCNumerics.cuh

```diff
@@ -14,102 +14,103 @@ struct THCNumerics {
-  static inline __host__ __device__ bool ne(int a, int b) { return a != b; }
-
-  static inline __host__ __device__  int neg(int a) { return -a; }
-  static inline __host__ __device__  int add(int a, int b) { return a + b; }
-  static inline __host__ __device__  int mul(int a, int b) { return a * b; }
-  static inline __host__ __device__  int sub(int a, int b) { return a - b; }
-  static inline __host__ __device__  int div(int a, int b) { return a / b; }
-  static inline __host__ __device__  int abs(int a) { return ::abs(a); }
+struct THCNumerics<int32_t> {
+  static inline __host__ __device__ int32_t min() { return INT_MIN; }
+  static inline __host__ __device__ int32_t max() { return INT_MAX; }
+
+  static inline __host__ __device__ bool lt(int32_t a, int32_t b) { return a < b; }
+  static inline __host__ __device__ bool le(int32_t a, int32_t b) { return a <= b; }
+  static inline __host__ __device__ bool gt(int32_t a, int32_t b) { return a > b; }
+  static inline __host__ __device__ bool ge(int32_t a, int32_t b) { return a >= b; }
+  static inline __host__ __device__ bool eq(int32_t a, int32_t b) { return a == b; }
+  static inline __host__ __device__ bool ne(int32_t a, int32_t b) { return a != b; }
+
+  static inline __host__ __device__  int32_t neg(int32_t a) { return -a; }
+  static inline __host__ __device__  int32_t add(int32_t a, int32_t b) { return a + b; }
+  static inline __host__ __device__  int32_t mul(int32_t a, int32_t b) { return a * b; }
+  static inline __host__ __device__  int32_t sub(int32_t a, int32_t b) { return a - b; }
+  static inline __host__ __device__  int32_t div(int32_t a, int32_t b) { return a / b; }
+  static inline __host__ __device__  int32_t abs(int32_t a) { return ::abs(a); }
 };
 
 template <>
-struct THCNumerics<long> {
-  static inline __host__ __device__ long min() { return LONG_MIN; }
-  static inline __host__ __device__ long max() { return LONG_MAX; }
-
-  static inline __host__ __device__ bool lt(long a, long b) { return a < b; }
-  static inline __host__ __device__ bool le(long a, long b) { return a <= b; }
-  static inline __host__ __device__ bool gt(long a, long b) { return a > b; }
-  static inline __host__ __device__ bool ge(long a, long b) { return a >= b; }
-  static inline __host__ __device__ bool eq(long a, long b) { return a == b; }
-  static inline __host__ __device__ bool ne(long a, long b) { return a != b; }
-
-  static inline __host__ __device__  long neg(long a) { return -a; }
-  static inline __host__ __device__  long add(long a, long b) { return a + b; }
-  static inline __host__ __device__  long mul(long a, long b) { return a * b; }
-  static inline __host__ __device__  long sub(long a, long b) { return a - b; }
-  static inline __host__ __device__  long div(long a, long b) { return a / b; };
-  static inline __host__ __device__  long abs(long a) { return labs(a); }
+struct THCNumerics<int64_t> {
+  static inline __host__ __device__ int64_t min() { return LONG_MIN; }
```

Comment:
https://msdn.microsoft.com/en-us/library/296az74e.aspx I just searched for the document is MSDN and that says the _I64_MIN and _I64_MAX should be used in MSVC.